api/tests/Feature/UserControllerTest.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\Jobs\UpdateWeatherJob;
+use App\Models\User;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Support\Facades\Bus;
+use Tests\TestCase;
+
+class UserControllerTest extends TestCase
+{
+    use RefreshDatabase;
+
+    public function test_get_users_with_weather()
+    {
+        Bus::fake();
+
+        User::factory()->create(['latitude' => 12.34, 'longitude' => 56.78]);
+
+        $response = $this->get('/users?limit=1');
+
+        $response->assertStatus(200);
+        $this->assertNull($response->json()[0]['weather']);
+        Bus::assertDispatched(UpdateWeatherJob::class);
+    }
+
+    public function test_get_users()
```

Comment:
You probably want to fake the `Bus` here too, even if you don't assert on it. Otherwise your jobs might execute synchronously which might potentially call some code you might not intend in the testing environment.